docs/LibraryEvolution.rst

```diff
@@ -1016,19 +994,16 @@ little. They allow the following changes:
 - The ``@discardableResult`` and ``@warn_unqualified_access`` attributes may
   be added to a method without any additional versioning information.
 
-Class and instance methods may be marked ``@inlineable``, with the same
+Class and instance methods may be marked ``@inlinable``, with the same
 restrictions as struct methods. Additionally, only non-overriding ``final``
-methods may be marked ``@inlineable``.
+methods may be marked ``@inlinable``.
 
 .. note::
 
     A previous draft of this document allowed non-``final`` methods to be
-    marked ``@inlineable``, permitting inlining based on speculative
-    devirtualization. This was removed both because of the added complexity for
-    users and because it would make methods on classes different from
-    struct/enum methods and top-level functions: a method on a class would be
-    part of the library's module interface even if it had been inlineable since
-    its introduction.
+    marked ``@inlinable``, permitting inlining based on speculative
```

Comment:
This restriction is not implemented today and the standard library has non-final inlinable class methods. Can we remove it?

(Note that `dynamic @inlinable` is still an error, since there's no way to statically call such a method.)